src/common/sha2/sha2_c.c

```diff
@@ -555,55 +563,63 @@ void oqs_sha2_sha512_inc_init_c(sha512ctx *state) {
 }
 
 void oqs_sha2_sha224_inc_ctx_clone_c(sha224ctx *stateout, const sha224ctx *statein) {
-	stateout->ctx = OQS_MEM_checked_malloc(PQC_SHA256CTX_BYTES);
-
+	stateout->ctx = OQS_MEM_malloc(PQC_SHA256CTX_BYTES);
+	if (!stateout->ctx) {
+		abort();
+	}
 	stateout->data_len = statein->data_len;
 	memcpy(stateout->data, statein->data, 128);
 	memcpy(stateout->ctx, statein->ctx, PQC_SHA256CTX_BYTES);
 }
 
 void oqs_sha2_sha256_inc_ctx_clone_c(sha256ctx *stateout, const sha256ctx *statein) {
-	stateout->ctx = OQS_MEM_checked_malloc(PQC_SHA256CTX_BYTES);
-
+	stateout->ctx = OQS_MEM_malloc(PQC_SHA256CTX_BYTES);
+	if (!stateout->ctx) {
+		abort();
+	}
 	stateout->data_len = statein->data_len;
 	memcpy(stateout->data, statein->data, 128);
 	memcpy(stateout->ctx, statein->ctx, PQC_SHA256CTX_BYTES);
 }
 
 void oqs_sha2_sha384_inc_ctx_clone_c(sha384ctx *stateout, const sha384ctx *statein) {
-	stateout->ctx = OQS_MEM_checked_malloc(PQC_SHA512CTX_BYTES);
-
+	stateout->ctx = OQS_MEM_malloc(PQC_SHA512CTX_BYTES);
+	if (!stateout->ctx) {
+		abort();
+	}
 	stateout->data_len = statein->data_len;
 	memcpy(stateout->data, statein->data, 128);
 	memcpy(stateout->ctx, statein->ctx, PQC_SHA512CTX_BYTES);
 }
 
 void oqs_sha2_sha512_inc_ctx_clone_c(sha512ctx *stateout, const sha512ctx *statein) {
-	stateout->ctx = OQS_MEM_checked_malloc(PQC_SHA512CTX_BYTES);
-
+	stateout->ctx = OQS_MEM_malloc(PQC_SHA512CTX_BYTES);
+	if (!stateout->ctx) {
+		abort();
+	}
 	stateout->data_len = statein->data_len;
 	memcpy(stateout->data, statein->data, 128);
 	memcpy(stateout->ctx, statein->ctx, PQC_SHA512CTX_BYTES);
 }
 
 /* Destroy the hash state. */
 void oqs_sha2_sha224_inc_ctx_release_c(sha224ctx *state) {
-	free(state->ctx); // IGNORE free-check
+	OQS_MEM_insecure_free(state->ctx);
 }
 
 /* Destroy the hash state. */
 void oqs_sha2_sha256_inc_ctx_release_c(sha256ctx *state) {
-	free(state->ctx); // IGNORE free-check
+	OQS_MEM_insecure_free(state->ctx);
```

Comment:
This is a question to @dstebila (as you added this 6 years ago): What's the rationale for having a function explicitly not checking for ptr==NULL, even labelling it "insecure"?

Reply:
And Yes, I know "free(NULL)" will not crash -- but I'd really like to understand the idea behind this function: Was it meant to contain some memory sanity check logic? Do we have to retain it? If not, this PR may be simpler....